stl/inc/sstream

```diff
@@ -100,6 +152,70 @@ public:
         _Init(_Newstr.c_str(), _Newstr.size(), _Mystate);
     }
 
+#if _HAS_CXX20
+    template <class _Alloc2, enable_if_t<_Is_allocator<_Alloc2>::value, int> = 0>
+    _NODISCARD basic_string<_Elem, _Traits, _Alloc2> str(const _Alloc2& _Al) const {
+        return basic_string<_Elem, _Traits, _Alloc2>{view(), _Al};
+    }
+
+    // The buffer cannot be moved to a string directly, because
+    // the buffer may already be full, and the terminating char is not '\0'.
+    // In that case, copy the string as usual.
+    _NODISCARD _Mystr str() && {
+        _Mystr _Result{_STD move(_Al)};

```

Comment:
The minimum size of buffer of stringbuf is defined as `_MINSIZE`. Right now, it is 32, which is larger than `_BUF_SIZE` (16). In addition, I've already added the logic that the buffer released from string will be at least larger than `_BUF_SIZE`. Therefore, it will avoid SSO, though it is a little tricky.

Reply:
Oh, considering of readonly buffer, maybe it cannot avoid SSO if the readonly string is small. Anyway I have to handle SSO...